app/src/org/commcare/android/tasks/ResourceEngineTask.java

```diff
@@ -149,16 +61,9 @@ public ResourceEngineTask(boolean upgradeMode, CommCareApp app,
     }
 
     @Override
-    protected ResourceEngineOutcomes doTaskBackground(String... profileRefs) {
+    protected AppInstallStatus doTaskBackground(String... profileRefs) {

```

Comment:
Could clean up this method even more by pulling the exception handling code for each separate exception out into methods.